src/internal/log/loggers.go

```diff
@@ -57,6 +64,41 @@ func SetGRPCLogLevel(l zapcore.Level) {
 	grpcLevel.SetLevel(l)
 }
 
+type atomicLeveler interface {
+	Level() zapcore.Level
+	SetLevel(zapcore.Level)
+}
+
+// revertLogLevel sets up a log level change that is reverted to an original level after the
+// duration, logging the revert with the text of msg, ensuring that even with multiple concurrent
+// calls, the most recent call cancels the actions of prior calls.
+func revertLogLevel(tPtr *atomic.Pointer[time.Timer], levelVar atomicLeveler, originalLevel, newLevel zapcore.Level, d time.Duration, msg string) {
+	wasSet := make(chan struct{})
+	oldTimer := tPtr.Swap(time.AfterFunc(d, func() {
+		<-wasSet
+		cur := levelVar.Level()
+		levelVar.SetLevel(originalLevel)
+		zap.L().Info(msg, zap.Stringer("from", cur), zap.Stringer("to", originalLevel))
+	}))
+	if oldTimer != nil {
+		oldTimer.Stop()
+	}
+	levelVar.SetLevel(newLevel)
+	close(wasSet) // prevent the revert from happening before we have the chance to set the new level
+}
+
+// SetLevelFor changes the global logger level for the set duration, and then reverts to the log
+// level at process startup.  Subsequent calls before expiration completely override previous calls;
+// the new level takes effect immediately and the previous scheduled revert is canceled.
+func SetLevelFor(l Level, d time.Duration) {
```

Comment:
Discussed this some offline, but I think these could be methods on an object.  Maybe a `LogManager`, which get's a reference to the logger setup by pachd in the pachd constructor, and exposes an API to manipulate it.

Reply:
Down below, the RPC is on the debug server, I think that's a good an object as any to own the timer and any state associated with reverting a level.

Reply:
I added a resettableLevel object. Much cleaner.  Also added extensive tests.